src/opencl/nvidia_specific.cl

```diff
@@ -0,0 +1,107 @@
+static inline void vec_copy(void *ret, const void *a, size_t num)
+{
+    limb_t *rp = (limb_t *)ret;
+    const limb_t *ap = (const limb_t *)a;
+    size_t i;
+
+    num /= sizeof(limb_t);
+
+    for (i = 0; i < num; i++)
+        rp[i] = ap[i];
+}
+
+inline void t_op_eq(struct vec256_t* x, const vec256 a) {

```

Comment:
some basic comments for the function signatures explaining what are the functions are doing would be really helpful. For example: `t_op_eq` was not clear if it was performing some equality check or assignment at the first glance. I had to read and understand the code of `vec_copy` for that (this is just a small example, it is getting more chaotic in more complex functions)